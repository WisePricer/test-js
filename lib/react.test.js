const renderer = require('react-test-renderer')
const { default: styled } = require('styled-components')

describe('react test setup', () => {
  it('allows testing React components with Enzyme matchers', () => {
    const TestComponent = () => React.createElement('div', {}, 'test content')
    const wrapper = shallow(React.createElement(TestComponent))
    expect(wrapper).toIncludeText('content')
  })

  it('includes styling in plain snapshots of styled components', () => {
    const StyledComponent = styled.div`
      color: red;
    `
    expect(renderer.create(React.createElement(StyledComponent, {}, 'test content')).toJSON()).toMatchSnapshot()
  })
})
