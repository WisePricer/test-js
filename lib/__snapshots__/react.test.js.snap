// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`react test setup includes styling in plain snapshots of styled components 1`] = `
.c0 {
  color: red;
}

<div
  className="c0"
>
  test content
</div>
`;
